include/re_types.h

```diff
@@ -258,3 +258,14 @@ typedef SSIZE_T ssize_t;
 #else
 #define re_restrict restrict
 #endif
+
+/* Socket helpers */
+#ifdef WIN32
+#define ERRNO_SOCK WSAGetLastError()
+#define BAD_SOCK INVALID_SOCKET
+typedef size_t re_sock_t;
+#else
+#define ERRNO_SOCK errno
+#define BAD_SOCK -1
+typedef int re_sock_t;
+#endif
```

Comment:
can you please make this hunk a separate PR ?

I also need it for UDP.

Please also add a `RE_` prefix to ERRNO_SOCK and BAD_SOCK